vespa-hadoop/src/main/java/com/yahoo/vespa/hadoop/pig/VespaDocumentOperation.java

```diff
@@ -182,12 +223,47 @@ public static String create(Operation op, String docId, Map<String, Object> fiel
     @SuppressWarnings("unchecked")
     private static void writeField(String name, Object value, Byte type, JsonGenerator g, Properties properties, Schema schema, Operation op, int depth) throws IOException {
         if (shouldWriteField(name, properties, depth)) {
-            g.writeFieldName(name);
-            if (shouldWritePartialUpdate(op, depth)) {
-                writePartialUpdate(value, type, g, name, properties, schema, op, depth);
-            } else {
-                writeValue(value, type, g, name, properties, schema, op, depth);
+            if (isPartialOperation(REMOVE_BAG_AS_MAP_FIELDS, name, properties) ||
+                    isPartialOperation(ADD_BAG_AS_MAP_FIELDS, name, properties)){
+                schema = (schema != null) ? schema.getField(0).schema : null;
```

Comment:
Extract line 228~257 into a new method. This could increase the readability of method, writeField.